sympy/combinatorics/coset_table.py

```diff
@@ -477,25 +511,49 @@ def coincidence(self, alpha, beta):
         A_dict_inv = self.A_dict_inv
         table = self.table
         p = self.p
-        l = 0
         # behaves as a queue
         q = []
-        self.merge(alpha, beta, q)
+        p_p = []
+        if modified:
+            p_p[beta] = self._grp.identity
+        l = 0
+        if modified:
+            self.modified_merge(alpha, beta, w, p_p, q)
```

Comment:
I think `p_p` should just be an attribute of `CosetTable`, like `P`. Then you wouldn't need to keep passing it to all these methods and some methods will look more like each other, e.g. `modified_rep` takes two arguments right now, but it could be taking just one, exactly like `rep`.